### PR TITLE
Move `platform` helm value out of `global`

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

**Comment by jwendell** [comment]
Not sure if we need a release note for this... It isn't a bug fix, or a feature :/

File: manifests/charts/istio-control/istio-discovery/templates/deployment.yaml [hunk 1]
```diff
@@ -165,7 +165,7 @@ spec:
               resourceFieldRef:
                 resource: limits.cpu
           - name: PLATFORM
-            value: "{{ .Values.global.platform }}"
+            value: "{{ .Values.platform }}"
```
Comment (hanxiaop) [inline manifests/charts/istio-control/istio-discovery/templates/deployment.yaml:168]: I think we need to temporarily support both values for backward compatibility?

File: manifests/charts/base/values.yaml [hunk 2]
```diff
@@ -36,3 +31,8 @@ defaults:
     enableIstioConfigCRDs: true
 
   defaultRevision: "default"
+
+# Platform where Istio is deployed. Possible values are: "openshift", "gcp".
+# An empty value means it is a vanilla Kubernetes distribution, therefore no special
+# treatment will be considered.
+platform: ""
```
Comment (hanxiaop) [inline manifests/charts/base/values.yaml:38]: I think this should also be placed under `defaults` to maintain consistency with the merge logic.

File: manifests/charts/istio-control/istio-discovery/templates/deployment.yaml (hunk 1 above)
Comment (howardjohn) [reply manifests/charts/istio-control/istio-discovery/templates/deployment.yaml:168]: I was kind of expecting to just have the gateway chart use `.Values.platform` and the rest continue using `global.platform`. This is cnsistent with some other fields I think.

The key principal is there is no unity of values between charts. `Values.foo.bar` in chart 1 is unrelated to `Values.foo.bar` in chart 2. Anything treating them the same -- like IstioOperator -- is flawed.

**Pull request closed without merging**